hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/s3/awssdk/S3SDKTestUtils.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.s3.awssdk;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.RandomAccessFile;
+import java.security.MessageDigest;
+import org.apache.commons.lang3.RandomUtils;
+import org.apache.hadoop.utils.InputSubstream;
+
+/**
+ * Utilities for S3 SDK tests.
+ */
+public final class S3SDKTestUtils {
+
+  private S3SDKTestUtils() {
+  }
+
+  public static byte[] calculateDigest(InputStream inputStream, int skip, int length) throws Exception {
+    int numRead;
+    byte[] buffer = new byte[1024];
+
+    MessageDigest complete = MessageDigest.getInstance("MD5");
+    if (skip > -1 && length > -1) {
+      inputStream = new InputSubstream(inputStream, skip, length);
```

Comment:
This inputStream needs to be closed. Looks like the parameters (skip and length) are always larger than -1.

Reply:
The responsibility of closing the InputStream is on the caller. Neither `calculateDigest` nor `InputSubstream` will close the InputStream.

I have added Javadoc to calculateDigest for clarity.

Reply:
No I'm sorry but this is not correct. The caller will not know the updated reference to the new input stream. The caller will close the old input stream but the new input stream will be leaked.

Reply:
Got it, thanks for the clarification. I created a new variable to hold reference of the `InputSubStream` (if the skip and length is specified) or the passed `InputSubstream`.  I also added `final` in the `InputStream` so that it's not updated.

Reply:
I created a POC to see what happens if a reference passed in a function is reassigned, seems the caller still refer to the original object.

```java
public class Example {
  private static class Foo {
    private int val;

    Foo(int val) {
      this.val = val;
    }

    public int getVal() {
      return val;
    }

  }

  private static void func(Foo foo) {
    foo = new Foo(100);
  }

  public static void main(String[] args) {
    Foo foo = new Foo(10);
    System.out.println(foo);
    System.out.println(foo.getVal());
    func(foo);
    System.out.println(foo.getVal());
    System.out.println(foo);
  }
}
```